src/main.rs

```diff
@@ -112,15 +113,21 @@ async fn main() -> anyhow::Result<()> {
         },
     ));
 
-    setup_logger()?;
-    let cli = Cli::parse();
-    match run(cli).await {
-        Ok(exit_code) => std::process::exit(exit_code),
-        Err(e) => {
-            log::error!("Error: {:?}", e);
-            std::process::exit(exitcode::SOFTWARE);
-        }
-    }
+    tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+        .block_on(async {
+            setup_logger().expect("Setup Logger");
```

Comment:
Nit: this would be a confusing error to see for a user. I think we can make it slightly better with a message like "unable to initialize". Also, do we want to hard fail on the logger not being able to be setup?

Reply:
We do not, your above comment helped me fix this one too :)